src/NHibernate/Dialect/Dialect.cs

```diff
@@ -215,17 +218,10 @@ public virtual string GetTypeName(SqlType sqlType)
 		{
 			if (sqlType.LengthDefined || sqlType.PrecisionDefined || sqlType.ScaleDefined)
 			{
-				string resultWithLength = _typeNames.Get(sqlType.DbType, sqlType.Length, sqlType.Precision, sqlType.Scale);
-				if (resultWithLength != null) return resultWithLength;
+				return _typeNames.Get(sqlType.DbType, sqlType.Length, sqlType.Precision, sqlType.Scale);
```

Comment:
Those `TypeNames` methods are never yielding `null`. (Unless a `Put` was done with `null`, but now `Put` throws null argument exception if it is attempted.)